components/ai_chat/core/browser/conversation_driver.cc

```diff
@@ -285,18 +286,11 @@ void ConversationDriver::InitEngine() {
 }
 
 bool ConversationDriver::HasUserOptedIn() {
-  base::Time last_accepted_disclaimer =
-        pref_service_->GetTime(ai_chat::prefs::kLastAcceptedDisclaimer);
-  return !last_accepted_disclaimer.is_null();
+  return ::ai_chat::HasUserOptedIn(pref_service_);
```

Comment:
This one and below doesn't really need to be in conversation_driver, but IMO it's better to keep it as is in this PR first.